### ISSUE TITLE
v1.1.0rc1 breaking API change: when should `training` be true in XGBoosterPredict(handle, dmat, option_mask, ntree_limit, training, ...)

### ISSUE THREAD

**Issue by glycerine** [dmlc/xgboost#5601]
I'm trying to update downstream libraries for v1.1.0rc1. There was a breaking change in the c_api.h to have an additional `training bool` parameter in this code change:

https://github.com/hcho3/xgboost/commit/f100b8d87871adaa693ea84548d69a598672d6f9
~~~
* Add `training` parameter for prediction method.
* [Breaking]: Add `training` to C API.
~~~

I'm looking at the patch but cannot tell when training should be true or false in the XGBoosterPredict() call. Could somebody offer some guidance?

I read that 
~~~
* \param training Whether the prediction value is used for training.
~~~

but that doesn't really tell me enough to understand what the consequences of training true (1) will be.

Thanks for you help.

**Comment by hcho3** [comment]
@glycerine The `training` argument is only relevant if you are using DART. Setting `training=true` will cause the prediction function to perform dropout, evaluating only a subset of the trees in the ensemble. Setting it to false will evaluate all trees.

**Comment by glycerine** [comment]
got it. Thanks @hcho3

**Comment by glycerine** [comment]
I suspect many maintainers of downstreams will have this question. It could be pre-empted by adding the above explanation to the documentation for the function.

I'll re-open this to be referenced if that is done. If that's not appropriate, feel free to close.

**Comment by hcho3** [comment]
@glycerine How common is it for downstream packages to directly access C API functions like `XGBoosterPredict()`?

**Comment by glycerine** [comment]
@hcho3 I would guess it is the most common API point. Python uses it, R uses it, Go uses it; I suspect any C libraries and anything using SWIG uses it since they can rarely use C++ due to lack of standard name mangling.
